infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -450,7 +444,14 @@ private void finishAllStartupTracesAsFailures(Throwable reason) {
    */
   private Function<Void, CompletionStage<Void>> bootstrap(
       List<CompletableFuture<?>> toCancelFutures, KubernetesMachineImpl machine) {
+    final Span activeSpan = tracerUtil.getActiveSpan();
     return ignored -> {
+      Span tracingSpan =
```

Comment:
Refactoring this with moving span creation code into a separate method would simplify this method. Since this method is about bootstrapping and not tracing it would benefit from moving tracing details out of its implementation details

Reply:
I'm not sure how it would be possible to move to a separate method, could you elaborate? I agree that it is a lot of cruft added to this method, but don't see a way around it.